internal/xds/translator/route.go

```diff
@@ -135,6 +140,31 @@ func buildXdsRouteAction(routeName string) *route.RouteAction {
 	}
 }
 
+func buildXdsWeightedRouteAction(httpRoute *ir.HTTPRoute) *route.RouteAction {
+	invalidBackendWeight := uint32(100 * float64(httpRoute.InvalidBackends) / float64(len(httpRoute.Destinations)+httpRoute.InvalidBackends))
+	destinationWeight := 100 - invalidBackendWeight
+	clusters := []*route.WeightedCluster_ClusterWeight{
```

Comment:
do we really need weighted clusters ? its another level of complexity or can we implement the same with https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/endpoint/v3/endpoint_components.proto#envoy-v3-api-msg-config-endpoint-v3-lbendpoint

Reply:
cc @lizan

Reply:
Where would the endpoint point to though?

Reply:
looks like this was already raised https://github.com/envoyproxy/envoy/issues/21930 and fixed via this route action https://github.com/envoyproxy/envoy/pull/22148 :)

Reply:
Yeah, thats exactly what I'm doing below. I updated the go control plane just so I could use that. I don't think it applies to endpoints though so thats why I went for the weighted cluster approach after reading that issue.

Reply:
My first impression here is do we really need to send invalid backends to Envoy? Is that required by Gateway API conformance?

Reply:
Yes, it's required for conformance that if input config is incorrect in certain ways, then 500 error codes (exactly) are returned for the traffic that matches that config, and if there are multiple match conditions, and one has an error, the proportions need to be correct, as @AliceProxy says above.

The idea here is that if you ask for an invalid config, that's what you get - otherwise you wouldn't be able to detect, for example, that a canary had a typo in its Service name until you flipped the weight to 100 - if we just silently dropped invalid config, it could get really weird.

Reply:
That makes sense if we don't have other way to give obvious feedback to user. In that case weighted cluster sounds reasonable to me.

Reply:
tried out empty endpoints
```
      - lb_endpoints:
        - endpoint:
            address: {}      
        loadBalancingWeight: 99 
        locality: {} 
``` 
which fails validation so looks like `this the way` !